### ISSUE TITLE
Compiler warns to remove parentheses producing invalid code

### ISSUE THREAD

**Issue by penelopezone** [rust-lang/rust#103435]
I tried this code:

```rust
fn main() {
     for (idx, rt_child) in root.iter_children().enumerate() {
          if let(Some(rt_child)) = rt_child {
            let reply_node = root_child_arena.get_reply_node(rt_child);
            info!("reply_node: {} {:?} {}", Move::from_index(idx), reply_node.get_visits(), reply_node.get_score());
          }
        }
}
```

I expected to see no compiler warning on the `if let(Some(rt_child)) = rt_child` line.

Instead, this happened: 
```

warning: unnecessary parentheses around pattern
 --> src/main.rs:3:17
  |
3 |           if let(Some(rt_child)) = rt_child {
  |                 ^              ^
  |
  = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
  |
3 -           if let(Some(rt_child)) = rt_child {
3 +           if letSome(rt_child) = rt_child {
  |

For more information about this error, try `rustc --explain E0425`.
warning: `playground` (bin "playground") generated 1 warning
error: could not compile `playground` due to 3 previous errors; 1 warning emitted
```

### Meta

here is a rust playground link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9d2bfd8156a76f0a565f0caf81b52d80
`rustc --version --verbose`:
```
1.64.0
```

**Comment by jruderman** [comment]
[Simpler](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=8fab660437393b643cb469b9936de123):

```rust
fn main() {
    if let(Some(_)) = Some(1) {
    }
}
```

which gives the error-introducing suggestion:

```
help: remove these parentheses
  |
2 -     if let(Some(_)) = Some(1) {
2 +     if letSome(_) = Some(1) {
```

**Comment by jruderman** [comment]
The warning is *correct*: it is unnecessary to put parens around the entire `if let` pattern. The suggestion just needs to include a space in this case, so it doesn't run the `let` and `Some` tokens together.

The suggestion should be:

```
help: remove these parentheses
  |
2 -     if let(Some(_)) = Some(1) {
2 +     if let Some(_) = Some(1) {
```

**Comment by chenyukang** [comment]
Another case:

```rust
fn main() {
    for(_x) in 1..10 { }
}
```

I guess we may have some other cases for `remove these parentheses`
